fix/forbid-unreachable.lua

```diff
@@ -0,0 +1,18 @@
+-- Fixes job cancellation spam when trying to use unreachable items.
+
+local count = 0
+for _,item in pairs(df.global.world.items.all) do
+    local pos = item.pos
+    local block = dfhack.maps.getTileBlock(pos)
+
+    if block then
+        local walkable = block.walkable[pos.x%16][pos.y%16]
```

Comment:
note that walkability is not quite the same as reachability, since a tile can be walkable (a floor) without being reachable (behind a closed drawbridge). For the purposes of this tool, though, I think the terminology and behavior as you have specified is perfectly fine. Maybe just call out this distinction in the docs.

Reply:
Added a note, good point.